src/routing_table/mod.rs

```diff
@@ -377,6 +398,37 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
+    // Returns the collection of groups to which a peer joining our group should connect.
+    //
+    // This will generally be all the groups in our routing table.  However, if the addition of the
+    // new node will cause our group to split, only the groups which will remain neighbours after
+    // the split are returned.  Returns `Err(Error::PeerNameUnsuitable)` if `name` is not within our
+    // group.
+    // TODO - should we check `name` isn't already in RT?
+    pub fn expect_add_to_our_group(&self, name: &T) -> Result<Groups<T>, Error> {
+        if !self.our_group_prefix.matches(name) {
+            return Err(Error::PeerNameUnsuitable);
+        }
+        let mut groups = self.groups.clone();
+        let mut our_group = unwrap!(groups.remove(&self.our_group_prefix));
+        if self.should_split_our_group(&our_group) {
```

Comment:
Shouldn't `name` be added to `our_group` before this check (and removed after it again)?